packages/query/src/stacks/bns/bns.utils.ts

```diff
@@ -1,37 +1,63 @@
 import { parseZoneFile } from '@fungible-systems/zone-file';
 import { BnsNamesOwnByAddressResponse } from '@stacks/stacks-blockchain-api-types';
+import axios from 'axios';
+import { getPrimaryName } from 'bns-v2-sdk';
+import { z } from 'zod';
 
+import { BNS_V2_API_BASE_URL, NetworkModes } from '@leather.io/models';
 import { isString, isUndefined } from '@leather.io/utils';
 
 import { StacksClient } from '../stacks-client';
+import { bnsV2NamesByAddressSchema } from './bns.schemas';
 
 /**
  * Fetch names owned by an address.
  */
 interface FetchNamesForAddressArgs {
-  client: StacksClient;
   address: string;
-  isTestnet: boolean;
+  network: NetworkModes;
   signal: AbortSignal;
 }
+
+type BnsV2NamesByAddressResponse = z.infer<typeof bnsV2NamesByAddressSchema>;
+
+async function fetchPrimaryName(address: string, network: NetworkModes) {
+  try {
+    const res = await getPrimaryName({ address, network });
+    return `${res?.name}.${res?.namespace}`;
+  } catch (error) {
+    // Ignore error
+    return undefined;
+  }
+}
+
 export async function fetchNamesForAddress({
-  client,
   address,
-  isTestnet,
   signal,
+  network,
 }: FetchNamesForAddressArgs): Promise<BnsNamesOwnByAddressResponse> {
-  const fetchFromApi = async () => {
-    return client.getNamesOwnedByAddress(address, signal);
-  };
-  if (isTestnet) {
-    return fetchFromApi();
+  const res = await axios.get<BnsV2NamesByAddressResponse>(
```

Comment:
A _nit_ suggestion here would be to:
- call `res` => `namesResponse`
- call `namesResponse` => `bnsNames`
- use a constant name `const addressHasMultipleNames = namesResponse.length > 1`
    - that way you can add your comment but it's in code not a comment